### DISCUSSION TITLE
Unexpected 400 when creating records via the API

### DISCUSSION THREAD

**Discussion by RickCastle2018** [pocketbase/pocketbase#5838] [Q&A] [answered]
Hi,

I recently faced a strange issue when creating records both with the Admin panel and when trying to use JS SDK / curl. Let's say I have a simple collection with name:string, creator:singleRelation (to the users collection). When I run the server locally with `go run main.go serve`, all my requests manually or via the client API pass through with 200 and are successful.

However, when I deploy my server with Docker Compose (I have a simple setup where it builds the server and has volume passing pb_data to the container), and use the same collections (before deployment I create a `collections_snapshot.go` file with `go run main.go migrate collections` and delete all the intermediary migrations created from the UI), every request to create a record via the API results in 400 errors. Additionally, the fact is that records are successfully created with Go methods in my additional routes without any problems, so this issue is only relevant to the PocketBase API that comes out of the box.

```
------WebKitFormBoundarypA1kgcvsU9esLMiC
Content-Disposition: form-data; name="name"

Test
------WebKitFormBoundarypA1kgcvsU9esLMiC
Content-Disposition: form-data; name="creatorId"

r741zaveyerj4p2
------WebKitFormBoundarypA1kgcvsU9esLMiC--

Response:
{
    "code": 400,
    "message": "Something went wrong while processing your request.",
    "data": {}
}
```

And that's what I see in the logs in the UI:

<img width="660" alt="image" src="https://github.com/user-attachments/assets/8fbc6f8f-43e9-4b26-84f7-7bcb08131076">

The same exact thing happens if I try to pass data manually with Curl as JSON in a POST request or use PocketBase JS SDK.

I wonder if you could suggest what could lead to such behavior of the system and where to look: unexpected end of JSON input error sounds weird when UI sends a POST Form-Data request...

Thanks,
Nick

**Comment by RickCastle2018** [comment]
Sorry to make this mistake twice creating an issue and not a discussion right away :)

**Comment by ganigeorgiev** [comment, marked as answer]
As mentioned in the rant in https://github.com/pocketbase/pocketbase/discussions/5825#discussioncomment-11196265, such posts don't have much chance resolving if there are no steps or details how to reproduce the issue.

The problem is most likely related either to your Docker setup, reverse proxy misconfiguration or caused by a conflicting hook, so double check your setup. It could be also caused by invalid inserted data, especially if you are using custom structs for persistence (`app.Dao()` doesn't run validations).

If you are still not able to identify the culprit try to provide more details what PocketBase version you are using and some steps to reproduce (_or ideally a **minimal** repo illustrating the problem_) and I'll try to investigate it.

**Comment by RickCastle2018** [comment]
Understood, thank you! It seems like a conflicting hook in my codebase - sorry to waste your time, running to a soon release, we must have missed something there - working with an additional database for analytics gathering (ClickHouse). Once again, thank you for support. I have not seen any other projects with such thorough and cool maintainers in my small, but somewhat an experience :)
